test/tests/functional/pbs_soft_walltime.py

```diff
@@ -0,0 +1,634 @@
+        Submit 3 jobs:
+        Job1 runs now with a soft_walltime=150
+        Job2 has a soft_walltime=150 and gets added to the calendar at now+150
+        Job3 has a walltime=300 and gets added to the calendar at now+300
+        Job4 gets added to the calendar at now+600
+        """
+        self.scheduler.set_sched_config({'strict_ordering': 'True'})
+        a = {'resources_available.ncpus': 1}
+        self.server.create_vnodes(name='vn', attrib=a, num=1, mom=self.mom)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {ATTR_backfill_depth: 3})
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        J1 = Job(TEST_USER, {'Resource_List.walltime': 300})
+        jid1 = self.server.submit(J1)
+        self.server.alterjob(jid1, {'Resource_List.soft_walltime': 150})
+
+        J2 = Job(TEST_USER, {'Resource_List.walltime': 300})
+        jid2 = self.server.submit(J2)
+        self.server.alterjob(jid2, {'Resource_List.soft_walltime': 150})
+
+        J3 = Job(TEST_USER, {'Resource_List.walltime': 300})
+        jid3 = self.server.submit(J3)
+
+        J4 = Job(TEST_USER, {'Resource_List.walltime': 300})
+        jid4 = self.server.submit(J4)
+
+        now = int(time.time())
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+
+        self.scheduler.log_match('Leaving Scheduling Cycle', starttime=now,
+                                 max_attempts=10)
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.compare_estiamtes(jid1, [jid2, jid3, jid4])
+
+    def test_filler_job(self):
+        """
+        Test to see if filler jobs will run based on their soft_walltime
+        Submit 3 jobs:
+        Job1 requests 1cpu and runs now
+        Job2 requests 2cpus gets added to the calendar at now+300
+        Job3 requests 1cpu and has a soft_walltime=150 and walltime=450
+        Job3 should run because its soft_walltime will finish before now+300
+        """
+        self.scheduler.set_sched_config({'strict_ordering': 'True'})
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes(name='vn', attrib=a, num=1, mom=self.mom)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        J1 = Job(TEST_USER, {'Resource_List.walltime': 300,
+                             'Resource_List.ncpus': 1})
+        jid1 = self.server.submit(J1)
+
+        J2 = Job(TEST_USER, {'Resource_List.walltime': 300,
+                             'Resource_List.ncpus': 2})
+        jid2 = self.server.submit(J2)
+
+        J3 = Job(TEST_USER, {'Resource_List.walltime': 450,
+                             'Resource_List.ncpus': 1})
+        jid3 = self.server.submit(J3)
+        self.server.alterjob(jid3, {'Resource_List.soft_walltime': 150})
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid3)
```

Comment:
please also check if job1 is running too

Reply:
Looks like I missed that.  Thanks for catching it.